manager/assets/modext/widgets/core/modx.window.js

```diff
@@ -150,31 +185,121 @@ MODx.Window = function(config) {
                     } else {
                         this.submit();
                     }
-
                 }
             ,scope: this
         }]
+        ,tools: [{
+            id: 'gear',
+            title: 'Window Settings',
+            // href: '#'
+            menu: {
+                xtype: 'menu',
+                anchor: true,
+                items: [
+                    {
+                    xtype: 'menucheckitem',
+                    text: 'Remove Masks',
+                    checked: true
+                    // bind: '{indented}'
+                }, {
+                    text: 'Disabled Item',
+                    disabled: true,
+                    separator: true
+                }
+                ]
+            }
+        }]
     });
     MODx.Window.superclass.constructor.call(this,config);
     this.options = config;
     this.config = config;
-
     this.addEvents({
         success: true
         ,failure: true
         ,beforeSubmit: true
     });
     this._loadForm();
-    this.on('show',function() {
-        if (this.config.blankValues) { this.fp.getForm().reset(); }
-        if (this.config.allowDrop) { this.loadDropZones(); }
-        this.syncSize();
-        this.focusFirstField();
-    },this);
-    this.on('afterrender', function() {
-        this.originalHeight = this.el.getHeight();
-        this.toolsHeight = this.originalHeight - this.body.getHeight() + 50;
-        this.resizeWindow();
+    this.on({
+        render: function() {
+            console.log('window render, this:', this);
+            if (MODx.config.enable_overlays) {
+                if (this.modxPseudoModal) {
+                    if (MODx.openPseudoModals.length === 0) {
+                        MODx.mask = this.container.createChild({cls:'ext-el-mask clickthrough'}, this.el.dom);
+                        MODx.mask.setStyle('backgroundColor', overlayCssColorNonblocking);
+                        // console.log('render, dynamic mask color: ', overlayCssColorNonblocking);
+                        MODx.mask.hide();
+                        MODx.mask.resizeMask = function() {
+                            // console.log('window resized!');
+                            MODx.mask.setSize(Ext.lib.Dom.getViewWidth(true), Ext.lib.Dom.getViewHeight(true));
+                        };
+                        // console.log('custom mask el: ', MODx.mask);
+                        window.addEventListener('resize', MODx.mask.resizeMask);
+                    }
+                    MODx.openPseudoModals.push({
+                        modalId: this.itemId
+                    });
+                    // console.log('open modxPseudoModals: ',MODx.openPseudoModals);
+                }
+                if (this.modal) {
+                    console.log('rendering real modal...');
```

Comment:
stray log, or something that still needs to be implemented? Should the pseudo modal handling just not run if this.modal is set?